app/controllers/search_controller.rb

```diff
@@ -35,12 +45,21 @@ def category
         order_by sort_arr.first, sort_arr.last
       end
     end
-    if category && name
+
+    if category && name && params['featured']
+      old_category_repos = Repository.where(category: category, featured: true).distinct
+      new_category_repos = Category.where(name: name).distinct.includes(:repository).map(&:repository).select{ |repo| repo[:featured] == true }
+      @repositories = (old_category_repos + new_category_repos).uniq.sort!{|a,b|b.updated_at <=> a.updated_at}.paginate(:per_page=>12,:page=>params[:page]) do
+        order_by :updated_at, :desc
+      end
+
+    elsif category && name
```

Comment:
Would it make more sense to do something like

```
@repositories = (@repositories1 + @repositories2).uniq
if params[:featured]
 @repositories = @repositories.select{|r| r.featured?}
end

@repositories.paginate....
```

Reply:
done